src/main/java/com/zaxxer/sansorm/internal/OrmWriter.java

```diff
@@ -313,18 +290,23 @@ public static int executeUpdate(Connection connection, String sql, Object... arg
       }
 
       stmt.executeUpdate();
+      fillGeneratedId(target, introspected, stmt, checkExistingId);
+   }
 
-      if (introspected.hasGeneratedId()) {
-         // Set auto-generated ID
-         final String idColumn = introspected.getIdColumnNames()[0];
-         ResultSet generatedKeys = stmt.getGeneratedKeys();
-         if (generatedKeys != null && generatedKeys.next()) {
-            introspected.set(target, idColumn, generatedKeys.getObject(1));
-            generatedKeys.close();
+   /** Sets auto-generated ID if not set yet */
+   private static <T> void fillGeneratedId(T target, Introspected introspected, PreparedStatement stmt, boolean checkExistingId) throws SQLException {
+      if (!introspected.hasGeneratedId()) {
+         return;
+      }
+      final String idColumn = introspected.getIdColumnNames()[0];
+      final Object idExisting = checkExistingId ? introspected.get(target, idColumn) : null;
+      if (idExisting == null || Integer.valueOf(0).equals(idExisting)) {
```

Comment:
Maybe it's too tied to our implementation, but we typically set the id column to ``-1`` on new objects (upon which we want an id to be generated).  This is probably in several hundred places in our application.

So maybe you could change the ``== 0`` comparison to the equivalent of ``<= 0``?

Reply:
okay. in the meantime could you please complete reivew for #15 and #17? :)

Reply:
thanks. now i have to resolve conflicts )

Reply:
@brettwooldridge, fixed, check it out